### PR TITLE
feat: partial solution for L1 -> L2 transactions table styling

### PR DESCRIPTION
**Pull request by coxmars**
### Description
This PR introduces a new page for displaying the latest L1 to L2 transactions. The new page includes a table that shows the following details for each transaction:
- L1 Block Number
- L2 Transaction Hash
- Timestamp
- L1 Transaction Hash
- L1 Transaction Origin
- Gas Limit

### Changes
- Created a new top-level route `/txs-enqueued` to display the L1 -> L2 transactions table.
- Added a new component `LatestL1L2TransactionsTable` that renders the transactions in a table format.
- Updated the `fetchLatestL1L2Transactions` function to return 50 placeholder transactions.
#### In Progress - I will improve this by tomorrow
- Included links and icons for external and internal navigation:
  - Links to block explorer for L1 block number, L1 transaction hash, and L1 transaction origin.
  - Internal links for L2 transaction hash.
- Adjusted table styling to ensure better readability and alignment of icons.

### Notes
- Placeholder data is used for development. Integration with the real API is in progress.
- This PR does not include pagination; it will be implemented in a separate issue.

### Screenshots
image

### CODE REVIEW

File: src/components/pages/txs/latest-l1-l2-transactions-table.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            <TableHead className="max-w-[10rem]">L1 Tx Hash</TableHead>
+            <TableHead className="max-w-[10rem]">L1 Tx Origin</TableHead>
+            <TableHead className="max-w-[8rem]">Gas Limit</TableHead>
+          </TableRow>
+        </TableHeader>
+        <TableBody>
+          {transactions.map((transaction) => (
+            <TableRow key={transaction.l2Hash}>
+              <TableCell className="max-w-[10rem] truncate">
+                <Link href={`https://explorer.l1.com/block/${transaction.l1BlockNumber}`}>
```
Comment (saimeunt) [inline src/components/pages/txs/latest-l1-l2-transactions-table.tsx]: Use the block explorer from the chain config in l1Chain defined here:
https://github.com/walnuthq/op-scan/blob/main/src/lib/chains.ts#L7

Use the proper styling as seen on the homepage: `text-primary hover:brightness-150`

**Review by saimeunt — CHANGES_REQUESTED** [review]
@coxmars This is a good start but it needs some polishing.

In particular, make sure you replicate styles from the homepage for the red links and show the icon when redirecting to the external L1 block explorer.

Screenshot 2024-06-30 at 00 05 17

**Comment by coxmars** [comment]
> @coxmars This is a good start but it needs some polishing.
> 
> In particular, make sure you replicate styles from the homepage for the red links and show the icon when redirecting to the external L1 block explorer.
> 
> Screenshot 2024-06-30 at 00 05 17
> 

Understood sir, I'm working on that so soon I will send the improvements 🫡

**Comment by coxmars** [comment]
> @coxmars This is a good start but it needs some polishing.
> 
> In particular, make sure you replicate styles from the homepage for the red links and show the icon when redirecting to the external L1 block explorer.
> 
> Screenshot 2024-06-30 at 00 05 17



> @coxmars This is a good start but it needs some polishing.
> 
> In particular, make sure you replicate styles from the homepage for the red links and show the icon when redirecting to the external L1 block explorer.
> 
> Screenshot 2024-06-30 at 00 05 17

Hi @saimeunt, I sent the improvements requested, thanks.

image

**Review by saimeunt — APPROVED** [review]

File: src/components/pages/txs/latest-l1-l2-transactions-table.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+                    <span className="truncate">{transaction.l1BlockNumber.toString()}</span>
+                    <SquareArrowOutUpRight className="ml-1 w-4 h-4 flex-shrink-0" />
+                  </a>
+                </div>
+              </TableCell>
+              <TableCell className="px-4 py-2 max-w-[14rem] truncate">
+                <div className="flex items-center gap-1 text-sm">
+                  <Link
+                    className="flex items-center truncate text-primary hover:brightness-150"
+                    href={`${l1Chain.blockExplorers.default.url}/tx/${transaction.l2Hash}`}
```
Comment (saimeunt) [inline src/components/pages/txs/latest-l1-l2-transactions-table.tsx]: The L2 hash is NOT an outside link so simply use `/tx/${transaction.l2Hash}` as href and remove the external icon here.

File: src/lib/utils.ts [hunk 3]
```diff
@@ -86,12 +86,16 @@ export const fetchTokensPrices = async () => {
 export const fetchLatestL1L2Transactions = async (): Promise<
```
Comment (saimeunt) [inline src/lib/utils.ts:39]: Fix the conflict in `@/lib/utils.ts` so this can be merged.

**Review by saimeunt — CHANGES_REQUESTED** [review]

File: src/lib/utils.ts (hunk 3 above)
Comment (coxmars) [reply src/lib/utils.ts:39]: > Fix the conflict in `@/lib/utils.ts` so this can be merged.

Hi @saimeunt, changes applied 🫡

**Review by saimeunt — APPROVED** [review]

**Pull request merged**